src/main/scala/strawman/collection/Set.scala

```diff
@@ -42,9 +58,13 @@ trait SetLike[A, +C[X] <: Set[X]]
 trait SetMonoTransforms[A, +Repr]
   extends IterableMonoTransforms[A, Repr] {
 
-  def & (that: Set[A]): Repr
+  def & (that: Set[A]): Repr = this.filter(that)
+
+}
+
+trait SetPolyTransforms[A, +C[X]] extends IterablePolyTransforms[A, C] {
 
-  def ++ (that: Set[A]): Repr
+  def ++ (that: Set[A]): C[A]
```

Comment:
Hmm, indeed I tried to build a counter-example but I failed: https://scastie.scala-lang.org/julienrf/KLo8JYnZRKmiIuolI7lZxg (also available in [this gist](https://gist.github.com/julienrf/7317162fec0f1a746c94021abd494771)). Scalac refuses to compile the unsound code :)

Still, I’m not 100% sure that it would be safe…